### DISCUSSION TITLE
Focus sidebar?

### DISCUSSION THREAD

**Discussion by eproxus** [NeoVintageous/NeoVintageous#891] [Q&A] [answered]
I'm using the trick described [here](https://blog.gerardroche.com/2023/05/18/neovintageous-reveal-in-side-bar/):

```vimrc
nnoremap <leader><leader> :RevealInSideBar<CR>
```

But it doesn't seem to focus the sidebar (to be able to e.g. add a new file next to the current, or rename it etc.). Is there a way to make the sidebar focused in the same remap?

**Comment by gerardroche** [comment]
I never noticed that because I use the neovintageous toggle sidebar command which wraps the built-in toggle sidebar command and enhances it by focusing the sidebar.

You can chain the two commands:

```
nnoremap <leader><leader> :RevealInSideBar<Bar>:FocusSideBar<CR>
```

The `<Bar>` special character let's you do some basic command chains. I wish I hadn't used `<Bar>` because it should be a pipe character `|`, the special `<Bar>` character should have been to allow you to specify a literal pipe character. It's going to be a backwards compatibility break when I change that later. 

You can the trick used by the toggle sidebar implementation to use the same mapping to switch the focus back from the side bar.

Menu → Preferences → Key Bindings:

```
    {
        "keys": [",", ","],
        "command": "focus_group",
        "args": {"group": 0},
        "context": [{"key": "control", "operand": "sidebar_tree"}]
    },
```

I've just tried this, it's really nice. Sometimes you just don't how useful something is until you're made aware of it.

I think I'll add a built-in command in neo like the toggle side bar:

```
nnoremap <leader><leader :Neovintageous action=reveal_side_bar<CR>
```

## Toggle sidebar 

To map a side bar toggle add a mapping: >

    nnoremap <leader>d :Neovintageous action=toggle_side_bar<CR>

Then add a key binding (this is for when the side bar has focus).

Menu → Preferences → Key Bindings: >

    {
        "keys": [",", "d"],
        "command": "neovintageous",
        "args": {"action": "toggle_side_bar"},
        "context": [{"key": "control", "operand": "sidebar_tree"}]
    }
 
See `:help nv`

**Comment by gerardroche** [comment, marked as answer]
The new reveal side bar command does several things that the builtin command in
sublime text does:

- Makes side bar visible if it's already visible
- Puts focus on the side bar

Here is an example using it in a mapping:

```vim
let mapleader=,
nnoremap <leader><leader> :Neovintageous action=reveal_side_bar<CR>
```

In order to use the same mapping to put focus back on the view create a keymap:

Menu → Preferences → Key Bindings

```js
{
    "keys": [",", ","],
    "command": "focus_group",
    "args": {"group": 0},
    "context": [{"key": "control", "operand": "sidebar_tree"}]
},
```

The reason you need the keymap definition is because it's specifically for when
the side bar has focus.  NeoVintageous can't handle that functionality directly.

See also the toggle side bar command: `:help nv`.

**Reply by eproxus** [reply]
That looks great. Is it released in Package Control already?

**Reply by gerardroche** [reply]
It's in the latest beta version. I will be releasing within the next few days.
